src/main/java/net/openhft/chronicle/queue/internal/util/InternalFileUtil.java

```diff
@@ -254,4 +268,25 @@ public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) {
             return FileVisitResult.CONTINUE;
         }
     }
+
+    private static class ProcFdWalkerWithPid extends ProcFdWalker {
+
+        private final Map<String, String> openFilesWithPid = new HashMap<>();
+
+        @Override
+        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
+            if (file.toAbsolutePath().toString().matches("/proc/\\d+/fd/\\d+")) {
+                try {
+                    final String e = file.toRealPath().toAbsolutePath().toString();
+                    final String pid = file.getName(1).toString(); // pid holding file open
```

Comment:
referencing folder by this hardcoded number worries me, if there's an option to save pids elsewhere then it breaks, maybe a more dynamic way of getting the file would be better?

Reply:
Not sure I agree there - it's no more hard-coded than the regex to check the file's naming structure. Are there Linux variants that use a different pathing structure for `/proc`?

If we expanded getAllOpenFIles to more operating systems (which would be keen) the methods used to get information start to vary significantly - on MacOS for example, we're parsing command-line output from lsof.

Reply:
can you use `file.getFileName().toString()`?